R/plotGMA.R

```diff
@@ -0,0 +1,270 @@
+#' plotGMA
+#' This function plots results of the Gene Methylation Analysis.
+#' It visualizes the number of hypo/hyper/dual methylated CpG
+#' sites in oncogenic mediators or in a user-supplied gene list. 
+#' The results are visualized either in a single heatmap or split 
+#' into different ones which is specified in the function's three 
+#' modes: split, complete and genelist.
+#' @param DEG_Methylation_Annotations A tibble which is outputted
+#' from the GMA function.
+#' @param Oncogenic_mediators_methylation_summary A tibble which
+#' is outputted from the GMA function.
+#' @param type A character string which can either be split,
+#' complete or genelist. If type is set to split, the entire
+#' dataset is split into groups of 40 genes and individual
+#' heatmaps of groups each containing 40 genes will be created and
+#' subsequently merged into one pdf where each page in the pdf
+#' is an individual heatmap. The genes will be sorted alphabetically. 
+#' If type is set to complete, a single heatmap is created where
+#' the number of differentially methylated CpG sites are shown for
+#' all oncogenic mediators. If type is set to genelist, a single
+#' heatmap will be created for genes supplied by the user in the
+#' genelist parameter. Default is split. 
+#' @param genelist A character string containing HUGO symbols
+#' of genes to be visualized in a single heatmap. Default is NULL.
+#' @param additionalFilename A character string that can be used
+#' to add a prefix or filepath to the filename of the pdf 
+#' visualizing the heatmap. Default is an empty string. 
```

Comment:
can we give it a more self-explanatory name, e.g. "filenamePrefix"

Reply:
yes we can do so. the other functions in Moonlight use the "additionalFilename" so that is why I have called it like that for consistency but I can update it here if it is better

Reply:
I've kept it as additionalFilename for consistency but let me know if I should change it here